tutorials/hentropy_search.ipynb

```diff
@@ -0,0 +1,454 @@
+    "    ba_l, ba_u = cfg.bounds_action\n",
+    "\n",
+    "    for i in range(cfg.acq_opt_iter):\n",
+    "        A_ = A.permute(1, 0, 2, 3)\n",
+    "        A_ = torch.sigmoid(A_) * (ba_u - ba_l) + ba_l\n",
+    "        posterior = qhes.model.posterior(A_)\n",
+    "        fantasized_outcome = qhes.action_sampler(posterior)\n",
+    "        # >>> n_fantasy_at_action_pts x n_fantasy_at_design_pts\n",
+    "        # ... x batch_size x num_actions x 1\n",
+    "\n",
+    "        fantasized_outcome = fantasized_outcome.squeeze(dim=-1)\n",
+    "        # >>> n_fantasy_at_action_pts x n_fantasy_at_design_pts\n",
+    "        # ... x batch_size x num_actions\n",
+    "\n",
+    "        losses = -qhes.loss_function(A=A_, Y=fantasized_outcome)\n",
+    "        # >>> n_fantasy_at_design_pts x batch_size\n",
+    "\n",
+    "        loss = losses.mean(dim=0).sum()\n",
+    "        A.grad = torch.autograd.grad(loss, A)[0]\n",
+    "        optimizer.step()\n",
+    "        optimizer.zero_grad()\n",
+    "\n",
+    "        if i % 200 == 0:\n",
+    "            print(f\"Eval optim round: {i}, Loss: {loss.item():.2f}\")\n",
+    "\n",
+    "    A = torch.sigmoid(A) * (ba_u - ba_l) + ba_l\n",
+    "    eval_metric = qhes.loss_function(A=A, Y=fantasized_outcome)\n",
+    "    eval_metric = eval_metric[0, 0].cpu().detach().numpy()\n",
+    "    optimal_action = A[0, 0].cpu().detach().numpy()\n",
+    "    value = qhes.loss_function(A=A, Y=func(A))[0, 0].cpu().detach().numpy()\n",
+    "    data_x = data.x.cpu().detach().numpy()\n",
+    "    next_x = next_x.cpu().detach().numpy()\n",
+    "\n",
+    "    # Plotting #########################################################################\n",
+    "    fig, ax = plt.subplots(1, 1)\n",
+    "    bounds_plot = cfg.bounds_design\n",
+    "    ax.set(xlabel=\"$x_1$\", ylabel=\"$x_2$\", xlim=bounds_plot, ylim=bounds_plot)\n",
+    "    title = \"$\\mathcal{H}_{\\ell, \\mathcal{A}}$-Entropy Search \" + cfg.task\n",
+    "    ax.set_title(label=title)\n",
+    "\n",
+    "    # Plot function in 2D ##############################################################\n",
+    "    X_domain, Y_domain = cfg.bounds_design, cfg.bounds_design\n",
+    "    n_space = 200\n",
+    "    X, Y = np.linspace(*X_domain, n_space), np.linspace(*Y_domain, n_space)\n",
+    "    X, Y = np.meshgrid(X, Y)\n",
+    "    XY = torch.tensor(np.array([X, Y]))\n",
+    "    Z = func(XY.reshape(2, -1).T).reshape(X.shape)\n",
+    "    cs = ax.contourf(X, Y, Z, levels=30, cmap=\"bwr\", alpha=0.7)\n",
+    "    ax.set_aspect(aspect=\"equal\")\n",
+    "    cbar = fig.colorbar(cs)\n",
+    "    cbar.ax.set_ylabel(\"$f(x)$\", rotation=270, labelpad=20)\n",
+    "\n",
+    "    # Plot data, optimal actions, next query ###########################################\n",
+    "    ax.scatter(data_x[:, 0], data_x[:, 1], label=\"Data\")\n",
+    "    ax.scatter(optimal_action[:, 0], optimal_action[:, 1], label=\"Action\")\n",
```

Comment:
The default orange color is quite hard to see on the red background - could you change the color here to make it more prominent?